app/Http/Controllers/Widgets/TopDevicesController.php

```diff
@@ -67,6 +68,11 @@ public function getView(Request $request)
         $settings = $this->getSettings();
         $sort = $settings['sort_order'];
 
+        // We use raw() function below, validate input and default to sane value.
+        if (! Validate::ascDesc($sort)) {
```

Comment:
This code for the default is going to be needed everywhere this is used... could add a default argument to the validate function.

Reply:
I didnt find another place but sure